account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -1007,26 +1009,38 @@ def _get_reconcile_line(
         )
         rates = []
         for vals in new_vals:
+            rate = False
             if vals["partner_id"] is False:
                 vals["partner_id"] = (False, self.partner_name)
-            reconcile_auxiliary_id, rate = self._compute_exchange_rate(
-                vals, line, reconcile_auxiliary_id
-            )
+            if vals.get("kind") not in ("suspense", "liquidity"):
+                reconcile_auxiliary_id, rate = self._compute_exchange_rate(
+                    vals, line, reconcile_auxiliary_id
+                )
             if rate:
                 rates.append(rate)
         new_vals += rates
         return reconcile_auxiliary_id, new_vals
 
     def _get_exchange_rate_amount(self, amount, currency_amount, currency, line):
-        return (
-            currency._convert(
+        if self.foreign_currency_id:
+            # take real rate of statement line to compute the exchange rate gain/loss
+            real_rate = self.amount / self.amount_currency
+            to_amount_journal_currency = currency_amount * real_rate
+            to_amount_company_currency = self.currency_id._convert(
+                to_amount_journal_currency,
+                self.company_id.currency_id,
+                self.company_id,
+                self.date,
+            )
+            to_amount = self.company_id.currency_id.round(to_amount_company_currency)
```

Comment:
In case of a foreign currency, we want to take the real rate (the one applied by the bank) instead of the one defined in Odoo. So we by pass the `currency._convert` in this case.